app/pages/lab/gold-standard-importer.cjsx

```diff
@@ -0,0 +1,120 @@
+React = require 'react'
+apiClient = require '../../api/client'
+counterpart = require 'counterpart'
+
+module.exports = React.createClass
+  IN_PROGRESS: 'IN_PROGRESS'
+  SUCCESSFUL: 'SUCCESSFUL'
+
+  getDefaultProps: ->
+    project: null
+    workflow: null
+    files: []
+    onClose: ->
+
+  getInitialState: ->
+    classificationsByFile: {}
+    errorsByFile: {}
+    progress: {}
+
+  componentDidMount: ->
+    @parseFiles @props.files
+
+  componentWillReceiveProps: (nextProps) ->
+    unless nextProps.files is @props.files
+      @parseFiles nextProps.files
+
+  parseFiles: (files) ->
+    @setState
+      classificationsByFile: {}
+      errorsByFile: {}
+      progress: null
+    Array::forEach.call files, (file) =>
+      reader = new FileReader
+      reader.onload = @handleFileRead.bind this, file
+      reader.readAsText file
+
+  handleFileRead: (file, e) ->
+    try
+      @state.classificationsByFile[file.name] = [].concat JSON.parse e.target.result
+      @setState classificationsByFile: @state.classificationsByFile
+    catch parseError
+      @state.errorsByFile[file.name] = parseError
+      @setState errorsByFile: @state.errorsByFile
+
+  startImport: ->
+    @setState
+      progress: {}
+    Object.keys(@state.classificationsByFile).forEach (fileName) =>
+      classifications = @state.classificationsByFile[fileName]
+      [Promise.resolve(), classifications...].reduce (awaitPreviousSave, classification, i1) =>
```

Comment:
If you supply the accumulator as a third arg to reduce here the index should be correct